app/src/pages/documentSearchResultsPage/DocumentSearchResultsPage.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useEffect, useState } from 'react';
+import { usePatientDetailsContext } from '../../providers/patientProvider/PatientProvider';
+import PatientSummary from '../../components/generic/patientSummary/PatientSummary';
+import { SearchResult } from '../../types/generic/searchResult';
+import DocumentSearchResults from '../../components/blocks/documentSearch/DocumentSearchResults';
+import { useNavigate } from 'react-router';
+import { routes } from '../../types/generic/routes';
+import { Link } from 'react-router-dom';
+import { SUBMISSION_STATE } from '../../types/pages/documentSearchResultsPage/types';
+import ProgressBar from '../../components/generic/progressBar/ProgressBar';
+import SpinnerButton from '../../components/generic/spinnerButton/SpinnerButton';
+import { Button } from 'nhsuk-react-components';
+import getDocumentSearchResults from '../../helpers/requests/documentSearchResults';
+import ServiceError from '../../components/layout/serviceErrorBox/ServiceErrorBox';
+
+import { useBaseAPIUrl } from '../../providers/configProvider/ConfigProvider';
+
+function DocumentSearchResultsPage() {
+    const [patientDetails] = usePatientDetailsContext();
+    const [searchResults, setSearchResults] = useState(Array<SearchResult>);
+    const [submissionState, setSubmissionState] = useState(SUBMISSION_STATE.INITIAL);
+    const [downloadState, setDownloadState] = useState(SUBMISSION_STATE.INITIAL);
+    const navigate = useNavigate();
+    const baseUrl = useBaseAPIUrl();
+
+    useEffect(() => {
+        if (!patientDetails?.nhsNumber) {
+            navigate(routes.HOME);
+        }
+
+        const search = async () => {
+            setSubmissionState(SUBMISSION_STATE.PENDING);
+            setSearchResults([]);
+
+            const nhsNumber: string = patientDetails?.nhsNumber || '';
+
+            try {
+                const results = await getDocumentSearchResults({
+                    nhsNumber,
+                    baseUrl,
+                });
+
+                if (results && results.length > 0) {
+                    setSearchResults(results);
+                }
+
+                setSubmissionState(SUBMISSION_STATE.SUCCEEDED);
+            } catch (e) {
+                setSubmissionState(SUBMISSION_STATE.FAILED);
+            }
+        };
+
+        void search();
+    }, [patientDetails, setSearchResults, setSubmissionState, navigate, baseUrl]);
+
+    const downloadAll = async () => {
+        setDownloadState(SUBMISSION_STATE.PENDING);
+
+        // Simulate delay for API call
+        // Temporary, to be replaced with real changes from PRMDR-110
+        setTimeout(() => {
+            setDownloadState(SUBMISSION_STATE.SUCCEEDED);
+        }, 2000);
+    };
+
+    return (
+        <>
+            <h1>Download electronic health records and attachments</h1>
+
+            {(submissionState === SUBMISSION_STATE.FAILED ||
+                downloadState === SUBMISSION_STATE.FAILED) && <ServiceError />}
+
```

Comment:
Sometimes when there are lots of boolean operators controlling the render of page parts, I make a list of constants at the top of the page something like:

const searchIsPending = submissionState === SUBMISSION_STATE
const searchHasFailed = submissionState === SUBMISSION_STATE.FAILED
const searchHasSucceeded = submissionState === SUBMISSION_STATE.SUCCESS
const searchIsFinished = searchHasFailed || searchHasSucceeded
const noResults = searchResults.length === 0

Reply:
It makes it much easier to digest the render conditions when we use these values in the render operators then. But i'd say this is optional more than required. Depends what you find easier to read